web/src/components/event-cards.js

```diff
@@ -111,59 +112,96 @@ class EventCards extends Component {
     }
   };
 
-  // Redirect to event detail page, When each card is clicked
+  /**
+   *  Redirect to event detail page, When each card is clicked
+   *  @param {int} eventId
+   */
   CardClicked = (eventId) => {
     return this.props.history.push({
       pathname: '/eventdetail',
       search: `?id=${eventId}`,
     });
   };
 
-  // Render of Event Cards
-  render() {
+  // Check Schedule Date of each Event
+  FilterSchedule = (cardIndex) => {
+    const nowDate = new Date();
+    let scheduleDate = null;
+    for(let scheduleIndex = 0;
+      scheduleIndex < this.props.events[cardIndex].schedule.length;
+      ++scheduleIndex){
+      scheduleDate = new Date(this.props.events[cardIndex].
+        schedule[scheduleIndex].startDateTime);
+      if(scheduleDate > nowDate){
+        return [scheduleIndex, <Datetime>{this.props.events[cardIndex].
+          schedule[scheduleIndex]}</Datetime>]
+      }
+    }
+    return [0, 'Ended']
```

Comment:
This seems a little dangerous. What if schedule[] is empty? Then the code below (caller of FilterSchedule) will try to access 0th element of schedule.